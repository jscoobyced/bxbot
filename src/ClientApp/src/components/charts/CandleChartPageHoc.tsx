import * as React from 'react';
import { CandleChartPage } from './CandleChartPage';
import { CandleChartDataService, ICandleChartDataService } from './CandleChartDataService';
import { CandleChartDataServiceMock } from './CandleChartDataServiceMock';
import { Pairing, CandleChartPageData } from './Models';

export class CandleChartPageHoc extends React.Component<{}, CandleChartPageData> {

    private readonly service: ICandleChartDataService;

    constructor(props: any, state: any) {
        super(props, state);
        const mode = process.env.mode;
        this.service = mode === 'development' ? new CandleChartDataServiceMock() : new CandleChartDataService();
        this.state = { pairings: [] }
    }

    public componentDidMount = () => {
        this.service.fetchCurrencyData(5)
            .then(data => {
                this.setState({ pairings: data });
            });
    }


    public render() {
        return <CandleChartPage pairings={this.state.pairings} />
    }
}