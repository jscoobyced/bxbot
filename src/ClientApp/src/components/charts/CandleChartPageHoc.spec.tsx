import { shallow } from 'enzyme';
import * as React from 'react';
import { TestUtils } from '../../../tests/TestUtils';
import { CandleChartPageHoc } from './CandleChartPageHoc';

test('CandleChartPageHoc snapshot for currency 5.', () => {
    window.fetch = TestUtils.mockFetch(
        '[{"timestamp":1,"low":1.0,"high":1.0,"open":1.0,"close":1.0,"current":1.0,"volume":0.0}]');
    const candleChartPageHoc = shallow(<CandleChartPageHoc />);
    expect(candleChartPageHoc).toMatchSnapshot();
});
