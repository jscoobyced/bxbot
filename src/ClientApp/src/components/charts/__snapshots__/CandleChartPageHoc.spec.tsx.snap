// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CandleChartPageHoc snapshot. 1`] = `
<article>
  <CandleChartCurrencySelector
    currencyOptions={Array []}
    onChangeCurrency={[Function]}
  />
  <CandleChartComponent
    currency=""
    loading={true}
    pairings={Array []}
  />
</article>
`;
